packages/flutter/lib/src/widgets/form.dart

```diff
@@ -56,8 +65,31 @@ class FormState extends State<Form> {
   int _generation = 0;
   Set<FormFieldState<dynamic>> _fields = new Set<FormFieldState<dynamic>>();
 
-  /// Called when a form field has changed. This will cause all form fields
-  /// to rebuild, useful if form fields have interdependencies.
+  @override
+  void dependenciesChanged() {
+    super.dependenciesChanged();
+    final ModalRoute<dynamic> route = ModalRoute.of(context);
+    if (route != null && route.isCurrent) {
+      // If it's the same ModalRoute, avoid adding our callback twice.
```

Comment:
I don't really understand this comment. There's no test anywhere that it's the same modal route...